<!DOCTYPE HTML>
    <head>
      <meta charset="utf-8"/>
      <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
      <meta name="viewport" content="width=device-width, initial-scale=1"/>
      <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags-->
      <title>AML VARAN | <?php echo $pageTitle ?></title>
      <link href="/var/www/common/stylesheets/reportstylesheet.css" rel="stylesheet"/>
    </head>
    <p class="head3">Clinical Variant Report</p>
    <p class="warningpdf">This is not a medical product. Use only for research purposes!</p><div style="width: 500px; float: right;"><b>Sample:</b><br/><table><tr><td style="width:130px">Sample taken:</td><td>2019-03-19</td></tr><tr><td>Diagnosis:</td><td>Demo sample</td></tr><tr><td>Comments:</td><td>Demo sample</td></tr></table><br/><br/></div><div style="width: 500px;"><b>Patient:</b><br/><table><tr><td style="width:130px">Name:</td><td>Demo patient</td></tr><tr><td>Birth date:</td><td>1972-02-28</td></tr><tr><td>Patient number:</td><td>1234</td></tr><tr><td>Sex:</td><td>M</td></tr></table><br/><br/></div><p class="head1pdf">Known Mutations</p>
        <p class="head2pdf">Mutations within the following regions are known to have therapeutical consequences.</p><div class="overview"><div style="overflow:hidden;"><p style="margin: 10px 0px 0px; color: #555;"><svg width=22px height=12px xmlns="http://www.w3.org/2000/svg"><rect x="0%" 10 height=12 rx="3" ry="3" width="100%" style="fill: #c61213"/></svg> Mutations found</p><div class="hotspotsbox mutationsfound" id="hotspotsbox5" style="float:left;">KIT (exons 8, 10, 11, 17)</div><div class="hotspotsbox mutationsfound" id="hotspotsbox6" style="float:left;">RUNX1 (exons 3-8)</div></div><div style="overflow:hidden;"><p style="margin: 10px 0px 0px; color: #555;"><svg width=22px height=12px xmlns="http://www.w3.org/2000/svg"><rect x="0%" 10 height=12 rx="3" ry="3" width="100%" style="fill: #dbab21"/></svg> No mutations found, but insufficient coverage</p><div class="hotspotsbox badcovered" id="hotspotsbox1" style="float:left;">NPM1 (type A)</div><div class="hotspotsbox badcovered" id="hotspotsbox2" style="float:left;">FLT3_TKD</div><div class="hotspotsbox badcovered" id="hotspotsbox3" style="float:left;">FLT3_ITD</div><div class="hotspotsbox badcovered" id="hotspotsbox4" style="float:left;">DNMT3A (R882)</div><div class="hotspotsbox badcovered" id="hotspotsbox7" style="float:left;">TET2 (exons 3-11)</div><div class="hotspotsbox badcovered" id="hotspotsbox8" style="float:left;">CEBPA</div><div class="hotspotsbox badcovered" id="hotspotsbox9" style="float:left;">IDH1_mut</div><div class="hotspotsbox badcovered" id="hotspotsbox10" style="float:left;">IDH2_mut</div><div class="hotspotsbox badcovered" id="hotspotsbox11" style="float:left;">ASXL1 (exons 9, 12)</div><div class="hotspotsbox badcovered" id="hotspotsbox12" style="float:left;">WT1 (rs16754)</div><div class="hotspotsbox badcovered" id="hotspotsbox13" style="float:left;">MLL_breakpt_clstr</div><div class="hotspotsbox badcovered" id="hotspotsbox14" style="float:left;">MLL_PTD_coding</div><div class="hotspotsbox badcovered" id="hotspotsbox15" style="float:left;">ABL1 (exons 4-6)</div><div class="hotspotsbox badcovered" id="hotspotsbox16" style="float:left;">JAK1_TKD</div><div class="hotspotsbox badcovered" id="hotspotsbox17" style="float:left;">JAK2 (Y931)</div><div class="hotspotsbox badcovered" id="hotspotsbox18" style="float:left;">BTK (exon15)</div><div class="hotspotsbox badcovered" id="hotspotsbox19" style="float:left;">MYD88_mut</div><div class="hotspotsbox badcovered" id="hotspotsbox20" style="float:left;">PLCG2 (Ser707Tyr)</div><div class="hotspotsbox badcovered" id="hotspotsbox21" style="float:left;">PML_muts</div><div class="hotspotsbox badcovered" id="hotspotsbox22" style="float:left;">SRSF2_c95ex1</div></div></div><table class="reporttable pdfwrapper" id="reporttable5"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside mutationsfound" id="hotspotsboxinside5">KIT (exons 8, 10, 11, 17)</div><p class="circlelabel mutationsfound">Mutations found in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="22.065877838184%" style="fill: #ffffff"/><defs><clipPath id="10">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="22.065877838184%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="22.065877838184%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="22.065877838184%" font-size="10">exon8</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="25.877198592901%" y="10" height="30" rx="10" ry="10" width="20.53086024944%" style="fill: #ffffff"/><defs><clipPath id="11">
            <rect x="25.877198592901%" y="10" height="30" rx="10" ry="10" width="20.53086024944%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="25.877198592901%" y="10" height="30" rx="10" ry="10" width="20.53086024944%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="25.877198592901%" y="65" width="20.53086024944%" font-size="10">exon10</text><line x1="26.477198592901%" y1="52" x2="26.477198592901%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="48.219379597058%" y="10" height="30" rx="10" ry="10" width="24.368404221298%" style="fill: #ffffff"/><defs><clipPath id="12">
            <rect x="48.219379597058%" y="10" height="30" rx="10" ry="10" width="24.368404221298%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="48.219379597058%" y="10" height="30" rx="10" ry="10" width="24.368404221298%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="48.219379597058%" y="65" width="24.368404221298%" font-size="10">exon11</text><line x1="48.819379597058%" y1="52" x2="48.819379597058%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="74.399104573073%" y="10" height="30" rx="10" ry="10" width="23.600895426927%" style="fill: #ffffff"/><defs><clipPath id="13">
            <rect x="74.399104573073%" y="10" height="30" rx="10" ry="10" width="23.600895426927%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="74.399104573073%" y="10" height="30" width="23.600895426927%" style="fill: #22670B" clip-path="url(#13)" /><rect x="74.399104573073%" y="10" height="30" rx="10" ry="10" width="23.600895426927%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="74.399104573073%" y="65" width="23.600895426927%" font-size="10">exon17</text><line x1="74.999104573073%" y1="52" x2="74.999104573073%" y2="43" style="stroke:grey;stroke-width:1" /><line x1="90.70866645347%" y1="39" x2="90.70866645347%" y2="11" style="stroke:#1DFFF5;stroke-width:2" /><line x1="90.40866645347%" y1="3" x2="90.70866645347%" y2="9" style="stroke:#1DFFF5;stroke-width:2" /><line x1="91.00866645347%" y1="3" x2="90.70866645347%" y2="9" style="stroke:#1DFFF5;stroke-width:2" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>KIT mutations were associated with a <b>higher cumulative incidence of relapse</b>, but in multivariate analysis this was only significant for cases with a higher mutant level of 25% or greater.

Mutations in exon 8 are usually in-frame insertions or deletions that affect an extracellular domain of the receptor implicated in receptor dimerization (KIT-ECD).</p><a href="http://www.nature.com/leu/journal/v27/n9/full/leu2013186a.html">http://www.nature.com/leu/journal/v27/n9/full/leu2013186a.html</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><table class="variantstable tablepdf"><thead>
  <tr><th>Color</th><th>chr</th><th>pos</th><th>dbSNP</th><th>Transcripts</th><th>Exons</th><th>Codons</th><th>Proteins</th><th>NRalt</th><th>Cvg</th><th>Freq</th><th>1000G</th><th>Provean</th><th>ClinVar</th><th>Cosmic</th><th>CosmicSNP</th><th>Category</th></tr>
</thead>
<tbody>
  <tr><td><div class='smallcircle' style='background-color: #1DFFF5;' ></div></td><td>4</td><td>55599321</td><td>rs121913507</td><td>ENST00000288135<br>ENST00000412167</td><td>17<br>17</td><td>c.2447A>T<br>c.2435A>T</td><td>p.D816V<br>p.D812V</td><td>257</td><td>558</td><td>0.4606</td><td></td><td>-7.53<br>-7.53</td><td>Uncertain significance</td><td></td><td></td><td></td></tbody>
            </table></td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable6"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside mutationsfound" id="hotspotsboxinside6">RUNX1 (exons 3-8)</div><p class="circlelabel mutationsfound">Mutations found in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="69.234322015525%" style="fill: #ffffff"/><defs><clipPath id="21">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="69.234322015525%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="69.234322015525%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="69.234322015525%" font-size="10">exon8</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="72.918532541841%" y="10" height="30" rx="10" ry="10" width="2.3318004504189%" style="fill: #ffffff"/><defs><clipPath id="20">
            <rect x="72.918532541841%" y="10" height="30" rx="10" ry="10" width="2.3318004504189%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="72.918532541841%" y="10" height="30" rx="10" ry="10" width="2.3318004504189%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="72.918532541841%" y="65" width="2.3318004504189%" font-size="10">exon7</text><line x1="73.518532541841%" y1="52" x2="73.518532541841%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="76.934543518575%" y="10" height="30" rx="10" ry="10" width="2.7636153486446%" style="fill: #ffffff"/><defs><clipPath id="19">
            <rect x="76.934543518575%" y="10" height="30" rx="10" ry="10" width="2.7636153486446%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="76.934543518575%" y="10" height="30" rx="10" ry="10" width="2.7636153486446%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="76.934543518575%" y="75" width="2.7636153486446%" font-size="10">exon6</text><line x1="77.534543518575%" y1="62" x2="77.534543518575%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="81.382369393536%" y="10" height="30" rx="10" ry="10" width="1.51135214379%" style="fill: #ffffff"/><defs><clipPath id="18">
            <rect x="81.382369393536%" y="10" height="30" rx="10" ry="10" width="1.51135214379%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="81.382369393536%" y="10" height="30" rx="10" ry="10" width="1.51135214379%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="81.382369393536%" y="65" width="1.51135214379%" font-size="10">exon5</text><line x1="81.982369393536%" y1="52" x2="81.982369393536%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="84.577932063642%" y="10" height="30" rx="10" ry="10" width="1.2267861258593%" style="fill: #ffffff"/><defs><clipPath id="15">
            <rect x="84.577932063642%" y="10" height="30" rx="10" ry="10" width="1.2267861258593%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="84.577932063642%" y="10" height="30" width="1.2267861258593%" style="fill: #22670B" clip-path="url(#15)" /><rect x="84.577932063642%" y="10" height="30" rx="10" ry="10" width="1.2267861258593%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="84.577932063642%" y="75" width="1.2267861258593%" font-size="10">rs587776810</text><line x1="85.177932063642%" y1="62" x2="85.177932063642%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="87.488928715817%" y="10" height="30" rx="10" ry="10" width="2.2598313007146%" style="fill: #ffffff"/><defs><clipPath id="16">
            <rect x="87.488928715817%" y="10" height="30" rx="10" ry="10" width="2.2598313007146%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="89.73436618659%" y="10" height="30" width="0.014393829940858%" style="fill: #22670B" clip-path="url(#16)" /><rect x="87.488928715817%" y="10" height="30" rx="10" ry="10" width="2.2598313007146%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="87.488928715817%" y="65" width="2.2598313007146%" font-size="10">exon4</text><line x1="88.088928715817%" y1="52" x2="88.088928715817%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="91.432970542847%" y="10" height="30" rx="10" ry="10" width="1.2267861258593%" style="fill: #ffffff"/><defs><clipPath id="17">
            <rect x="91.432970542847%" y="10" height="30" rx="10" ry="10" width="1.2267861258593%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="91.432970542847%" y="10" height="30" width="1.2267861258593%" style="fill: #22670B" clip-path="url(#17)" /><rect x="91.432970542847%" y="10" height="30" rx="10" ry="10" width="1.2267861258593%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="91.432970542847%" y="75" width="1.2267861258593%" font-size="10">rs587776809</text><line x1="92.032970542847%" y1="62" x2="92.032970542847%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="94.343967195022%" y="10" height="30" rx="10" ry="10" width="3.6560328049778%" style="fill: #ffffff"/><defs><clipPath id="14">
            <rect x="94.343967195022%" y="10" height="30" rx="10" ry="10" width="3.6560328049778%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="94.343967195022%" y="10" height="30" width="3.6560328049778%" style="fill: #90191C" clip-path="url(#14)" /><rect x="94.343967195022%" y="10" height="30" rx="10" ry="10" width="3.6560328049778%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="94.343967195022%" y="65" width="3.6560328049778%" font-size="10">exon3</text><line x1="94.943967195022%" y1="52" x2="94.943967195022%" y2="43" style="stroke:grey;stroke-width:1" /><line x1="96.99243190414%" y1="39" x2="96.99243190414%" y2="11" style="stroke:#1DFFF5;stroke-width:2" /><line x1="96.69243190414%" y1="3" x2="96.99243190414%" y2="9" style="stroke:#1DFFF5;stroke-width:2" /><line x1="97.29243190414%" y1="3" x2="96.99243190414%" y2="9" style="stroke:#1DFFF5;stroke-width:2" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>RUNX1 mutations are frequent in de novo AML with noncomplex karyotype and confer an unfavorable prognosis</p><a href="http://www.ncbi.nlm.nih.gov/pubmed/21148331">http://www.ncbi.nlm.nih.gov/pubmed/21148331</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><table class="variantstable tablepdf"><thead>
  <tr><th>Color</th><th>chr</th><th>pos</th><th>dbSNP</th><th>Transcripts</th><th>Exons</th><th>Codons</th><th>Proteins</th><th>NRalt</th><th>Cvg</th><th>Freq</th><th>1000G</th><th>Provean</th><th>ClinVar</th><th>Cosmic</th><th>CosmicSNP</th><th>Category</th></tr>
</thead>
<tbody>
  <tr><td><div class='smallcircle' style='background-color: #1DFFF5;' ></div></td><td>21</td><td>36259324</td><td>rs111527738</td><td>ENST00000300305<br>ENST00000325074<br>ENST00000344691<br>ENST00000358356<br>ENST00000399237<br>ENST00000399240<br>ENST00000437180<br>ENST00000455571<br>ENST00000486278</td><td>3<br>2<br>1<br>1<br>2<br>1<br>4<br>3<br>4</td><td>c.167T>C<br>c.131T>C<br>c.86T>C<br>c.86T>C<br>c.131T>C<br>c.86T>C<br>c.167T>C<br>c.128T>C<br>c.95T>C</td><td>p.L56S<br>p.L44S<br>p.L29S<br>p.L29S<br>p.L44S<br>p.L29S<br>p.L56S<br>p.L43S<br>p.L32S</td><td>440</td><td>453</td><td>0.9713</td><td>0.0103834</td><td>-0.65<br>-0.54<br>-0.08<br>-0.09<br>0.4<br>-0.8<br>-0.1<br>0.31</td><td>Benign/Likely benign</td><td></td><td></td><td></td></tbody>
            </table></td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable1"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside1">NPM1 (type A)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="2">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" width="96%" style="fill: #22670B" clip-path="url(#2)" /><rect x="5.2178770949721%" y="10" height="30" width="1.608938547486%" style="fill: #22670B" clip-path="url(#2)" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">exon12</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="5.4824343555543%" y="10" height="30" width="1.4385132684097%" style="fill: #ffffff"/><rect x="5.4824343555543%" y="10" height="30" width="1.4385132684097%" style="fill: #22670B"/><rect x="5.4824343555543%" y="10" height="30" width="1.4385132684097%" style="fill: #22670B"/><rect x="5.4824343555543%" y="10" height="30" width="1.4385132684097%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="5.4824343555543%" y="75" width="1.4385132684097%" font-size="10">type A</text><line x1="6.0824343555543%" y1="62" x2="6.0824343555543%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>NPM1 exon 12 mutations are frequently identified in patients with cytogenetically normal acute myeloid leukemia (AML) and often co-occur with FLT3-ITD. FLT3 status should also be evaluated as co-occurence with FLT3-ITD may impact prognosis. 
Exon 12 mutations have been identified as a predictor of good prognostic outcomes in the absence of FLT3-ITD.

Due to their high frequency, NPM1 mutations have been retrospectively analyzed in the context of a number of therapies including variable results following ATRA treatment as well as improved response to high-dose daunorubicin or valproic acid.

Additionally, multiple groups have shown increased surface expression of CD33 associated with NPM1 mutation, suggesting these patients may respond to anti-CD33 therapy. Cytoplasmic sequestration of NPM1 (NPM1c) is associated with a good response to induction therapy.</p><a href="https://civicdb.org/events/genes/35/summary/variants/86/summary#variant">https://civicdb.org/events/genes/35/summary/variants/86/summary#variant</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable2"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside2">FLT3_TKD</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="4">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">exon20</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="28.536585365854%" y="10" height="30" width="3.9024390243902%" style="fill: #ffffff"/><rect x="28.536585365854%" y="10" height="30" width="3.9024390243902%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="28.536585365854%" y="65" width="3.9024390243902%" font-size="10">D835/I836</text><line x1="29.136585365854%" y1="52" x2="29.136585365854%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>FLT3-TKD mutations are small mutations in the activation loop of FLT3, mostly representing point mutations in codon D835 or deletions of codon I836 (on exon 20 [Myeloid Leukemia: Methods and Protocols, Iland]). They induce constitutive tyrosine phosphorylation leading to activation of the receptor tyrosine kinase and are supposed to represent gain-of-function mutations.</p><a href="http://www.bloodjournal.org/content/111/5/2527.long">http://www.bloodjournal.org/content/111/5/2527.long</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable3"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside3">FLT3_ITD</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="5">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">exon14-15</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>FLT3-ITD (internal tandem duplications) frequently occur in patients with hematologic malignancies such as chronic myelogenous leukemia, acute myeloid leukemia (AML) and myelodysplastic syndrome, but particularly in cytogenetically normal AML (CN-AML). These duplication events disrupt the juxtamembrane domain of FLT3 and can be the result of a duplication of internal FLT3 sequence or other unrelated sequence resulting in an in-frame duplication event. The length of these duplications can vary widely which may have prognostic consequences, but this has not been conclusively determined. FLT3-ITD mutations overall have generally been associated with poor prognosis. Additional genes associated with CN-AML such as NPM1 may modulate the prognosis associated with this variant.</p><a href="https://civicdb.org/events/genes/24/summary/variants/55/summary#variant">https://civicdb.org/events/genes/24/summary/variants/55/summary#variant</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable4"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside4">DNMT3A (R882)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="92.743348680397%" style="fill: #ffffff"/><defs><clipPath id="8">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="92.743348680397%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="91.567206602301%" y="10" height="30" width="0.25409136624766%" style="fill: #90191C" clip-path="url(#8)" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="92.743348680397%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="92.743348680397%" font-size="10">exon23</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="90.484371980837%" y="10" height="30" width="1.3742983784267%" style="fill: #ffffff"/><rect x="90.484371980837%" y="10" height="30" width="1.3742983784267%" style="fill: #22670B"/><rect x="90.484371980837%" y="10" height="30" width="1.3742983784267%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="90.484371980837%" y="65" width="1.3742983784267%" font-size="10">R882</text><line x1="91.084371980837%" y1="52" x2="91.084371980837%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="96.625701621573%" y="10" height="30" rx="10" ry="10" width="1.3742983784267%" style="fill: #ffffff"/><defs><clipPath id="9">
            <rect x="96.625701621573%" y="10" height="30" rx="10" ry="10" width="1.3742983784267%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="96.625701621573%" y="10" height="30" width="1.3742983784267%" style="fill: #22670B" clip-path="url(#9)" /><rect x="96.625701621573%" y="10" height="30" rx="10" ry="10" width="1.3742983784267%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="96.625701621573%" y="75" width="1.3742983784267%" font-size="10">rs796065342</text><line x1="97.225701621573%" y1="62" x2="97.225701621573%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>DNMT3A mutations can lead to epigenetic modifications and can cause excessive hypermethylation of tumor suppressor genes resulting in increased tumor growth. Methyltransferase inhibitors including decitabine and azacitidine inhibit the activity of DNMT3A and will decrease DNA methylation and potentially restore regular transcriptional activity of tumor suppressor genes.</p><a href="https://civicdb.org/events/genes/18/summary/variants/189/summary#variant">https://civicdb.org/events/genes/18/summary/variants/189/summary#variant</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable7"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside7">TET2 (exons 3-11)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="30.141343709245%" style="fill: #ffffff"/><defs><clipPath id="22">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="30.141343709245%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="30.141343709245%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="30.141343709245%" font-size="10">exon3</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="33.796516123038%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="fill: #ffffff"/><defs><clipPath id="23">
            <rect x="33.796516123038%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="33.796516123038%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="33.796516123038%" y="65" width="1.2220984467466%" font-size="10">exon4</text><line x1="34.396516123038%" y1="52" x2="34.396516123038%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="36.673786983578%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="fill: #ffffff"/><defs><clipPath id="24">
            <rect x="36.673786983578%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="36.673786983578%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="36.673786983578%" y="75" width="1.2220984467466%" font-size="10">exon5</text><line x1="37.273786983578%" y1="62" x2="37.273786983578%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="39.551057844117%" y="10" height="30" rx="10" ry="10" width="1.8233113850166%" style="fill: #ffffff"/><defs><clipPath id="25">
            <rect x="39.551057844117%" y="10" height="30" rx="10" ry="10" width="1.8233113850166%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="39.551057844117%" y="10" height="30" rx="10" ry="10" width="1.8233113850166%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="39.551057844117%" y="65" width="1.8233113850166%" font-size="10">exon6</text><line x1="40.151057844117%" y1="52" x2="40.151057844117%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="43.029541642927%" y="10" height="30" rx="10" ry="10" width="1.3173206657297%" style="fill: #ffffff"/><defs><clipPath id="26">
            <rect x="43.029541642927%" y="10" height="30" rx="10" ry="10" width="1.3173206657297%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="43.029541642927%" y="10" height="30" rx="10" ry="10" width="1.3173206657297%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="43.029541642927%" y="75" width="1.3173206657297%" font-size="10">exon7</text><line x1="43.629541642927%" y1="62" x2="43.629541642927%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="46.00203472245%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="fill: #ffffff"/><defs><clipPath id="27">
            <rect x="46.00203472245%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="46.00203472245%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="46.00203472245%" y="65" width="1.2220984467466%" font-size="10">exon8</text><line x1="46.60203472245%" y1="52" x2="46.60203472245%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="48.87930558299%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="fill: #ffffff"/><defs><clipPath id="28">
            <rect x="48.87930558299%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="48.87930558299%" y="10" height="30" rx="10" ry="10" width="1.2220984467466%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="48.87930558299%" y="75" width="1.2220984467466%" font-size="10">exon9</text><line x1="49.47930558299%" y1="62" x2="49.47930558299%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="51.756576443529%" y="10" height="30" rx="10" ry="10" width="3.0970121611525%" style="fill: #ffffff"/><defs><clipPath id="29">
            <rect x="51.756576443529%" y="10" height="30" rx="10" ry="10" width="3.0970121611525%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="51.756576443529%" y="10" height="30" rx="10" ry="10" width="3.0970121611525%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="51.756576443529%" y="65" width="3.0970121611525%" font-size="10">exon10</text><line x1="52.356576443529%" y1="52" x2="52.356576443529%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="56.508761018475%" y="10" height="30" rx="10" ry="10" width="41.491238981525%" style="fill: #ffffff"/><defs><clipPath id="30">
            <rect x="56.508761018475%" y="10" height="30" rx="10" ry="10" width="41.491238981525%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="56.508761018475%" y="10" height="30" rx="10" ry="10" width="41.491238981525%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="56.508761018475%" y="75" width="41.491238981525%" font-size="10">exon11</text><line x1="57.108761018475%" y1="62" x2="57.108761018475%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>In patients receiving intensive therapy, one group found a negative prognostic effect restricted to patients with CN-AML and favourable genetic aberrations (NPM1 mutated/FLT3 no ITD, or CEBPA mutated) (Metzeler et al, 2012b), which could not be confirmed in other analyses (Shen et al, 2011; Gaidzik et al, 2012; Nomdedeu et al, 2012). 
Another group found TET2 mutations were a negative prognostic factor in patients with intermediate cytogenetic risk (Chou et al, 2011a). However, it lost its independent significance when additional genetic aberrations were considered (Hou et al, 2012).

Higher response rate after 5-azacitidine in patients with 20–30% BM blasts [Itzykson et al (2011)]</p><a href="http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full">http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable8"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside8">CEBPA</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="33">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">exon1</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="56.835970667696%" y="10" height="30" width="5.9282130451563%" style="fill: #ffffff"/><rect x="56.835970667696%" y="10" height="30" width="5.9282130451563%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="56.835970667696%" y="65" width="5.9282130451563%" font-size="10">bZIP</text><line x1="57.435970667696%" y1="52" x2="57.435970667696%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="80.919336163643%" y="10" height="30" width="13.005017367812%" style="fill: #ffffff"/><rect x="80.919336163643%" y="10" height="30" width="13.005017367812%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="80.919336163643%" y="65" width="13.005017367812%" font-size="10">NTerm</text><line x1="81.519336163643%" y1="52" x2="81.519336163643%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>Importantly, recent studies show that only double, but not single, CEBPA mutations predict for this <b>favorable outcome</b>, whereas AML with a single CEBPA mutation is associated with survival similar to that of AML with wild-type CEBPA.
Therefore, with regard to disease classification and risk stratification, <b>only AML with double CEBPA mutations</b> likely define a distinct entity with <b>favorable prognosis</b></p><a href="http://jco.ascopubs.org/content/29/5/475.full#sec-3">http://jco.ascopubs.org/content/29/5/475.full#sec-3</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable9"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside9">IDH1_mut</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="35">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">exon4</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="34.17550274223%" y="10" height="30" width="1.423522242535%" style="fill: #ffffff"/><rect x="34.17550274223%" y="10" height="30" width="1.423522242535%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="34.17550274223%" y="65" width="1.423522242535%" font-size="10">rs11554137</text><line x1="34.77550274223%" y1="52" x2="34.77550274223%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>Mutations of the isocitrate dehydrogenase 1 gene (IDH1) have been reported as an adverse prognostic factor only in genetically defined subgroups of patients (Abbas et al, 2010; Boissel et al, 2010; Green et al, 2010b; Marcucci et al, 2010; Paschka et al, 2010; Schnittger et al, 2010; Nomdedeu et al, 2012), while a prognostic impact could not be detected by others (Chou et al, 2010a; Ho et al, 2010; Wagner et al, 2010; Shen et al, 2011; Zhang et al, 2011; Chotirat et al, 2012; Patel et al, 2012; Ravandi et al, 2012)</p><a href="http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full">http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable10"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside10">IDH2_mut</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="37">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">exon4</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="12.732919254658%" y="10" height="30" width="2.3850931677019%" style="fill: #ffffff"/><rect x="12.732919254658%" y="10" height="30" width="2.3850931677019%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="12.732919254658%" y="65" width="2.3850931677019%" font-size="10">R172</text><line x1="13.332919254658%" y1="52" x2="13.332919254658%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>The prognostic effect of IDH2 mutations is even less clear, with some groups finding <b>a positive impact on survival</b> (Chou et al, 2011b; Patel et al, 2012), while others found <b>a negative impact</b> in defined subgroups (Paschka et al, 2010; Rockova et al, 2011; Ravandi et al, 2012) or in the total cohort when only the R172 mutation was evaluated (Boissel et al, 2010). 
One group found divergent results with a favourable impact of the R140 and a nonsignificant unfavourable impact of the R172 mutation (Green et al, 2011).
 Most groups did not find a prognostic impact of IDH2 mutations (Abbas et al, 2010; Marcucci et al, 2010; Thol et al, 2010; Chotirat et al, 2012; Nomdedeu et al, 2012; Renneville et al, 2012). A recent meta-analysis found a positive prognostic impact of IDH2 mutations on survival (Zhou et al, 2012).</p><a href="http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full">http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable11"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside11">ASXL1 (exons 9, 12)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="19.342473831638%" style="fill: #ffffff"/><defs><clipPath id="38">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="19.342473831638%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="19.342473831638%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="19.342473831638%" font-size="10">exon9</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="23.224826772814%" y="10" height="30" rx="10" ry="10" width="74.775173227186%" style="fill: #ffffff"/><defs><clipPath id="39">
            <rect x="23.224826772814%" y="10" height="30" rx="10" ry="10" width="74.775173227186%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="23.224826772814%" y="10" height="30" rx="10" ry="10" width="74.775173227186%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="23.224826772814%" y="65" width="74.775173227186%" font-size="10">exon12</text><line x1="23.824826772814%" y1="52" x2="23.824826772814%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>ASXL1 mutations were found to be an independent adverse prognostic factor for survival in patients with CN-AML (Schnittger et al, 2013) and in older patients with CN-AML and favourable genetic risk (Metzeler et al, 2011). ASXL1 mutations were an adverse factor for survival in a univariate analysis, but not in a multivariate analysis in a Taiwanese study (Chou et al, 2010b).</p><a href="http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full">http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable12"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside12">WT1 (rs16754)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="33.284422073218%" style="fill: #ffffff"/><defs><clipPath id="42">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="33.284422073218%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="33.284422073218%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="33.284422073218%" font-size="10">exon9</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="37.166775014395%" y="10" height="30" rx="10" ry="10" width="60.833224985605%" style="fill: #ffffff"/><defs><clipPath id="41">
            <rect x="37.166775014395%" y="10" height="30" rx="10" ry="10" width="60.833224985605%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="37.166775014395%" y="10" height="30" rx="10" ry="10" width="60.833224985605%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="37.166775014395%" y="65" width="60.833224985605%" font-size="10">exon7</text><line x1="37.766775014395%" y1="52" x2="37.766775014395%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="93.43329804293%" y="10" height="30" width="1.3967569977154%" style="fill: #ffffff"/><rect x="93.43329804293%" y="10" height="30" width="1.3967569977154%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="93.43329804293%" y="65" width="1.3967569977154%" font-size="10">rs16754</text><line x1="94.03329804293%" y1="52" x2="94.03329804293%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>Wilms tumor 1 gene (WT1) mutations are detected in approximately 10–13% of AML and an independent <b>negative prognostic impact</b> (Paschka et al, 2008; Virappane et al, 2008), as well as <b>no prognostic impact</b> (Gaidzik et al, 2009; Damm et al, 2010) have been reported in CN-AML patients.</p><a href="http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full">http://onlinelibrary.wiley.com/doi/10.1111/bjh.12750/full</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable13"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside13">MLL_breakpt_clstr</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="43">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">MLL_breakpt_clstr</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>MLL-PTD have been associated with <b>inferior CR duration and relapse-free survival</b>, although <b>more recent studies show no prognostic impact</b> in patients with CN-AML intensively treated with autologous HSCT or four cycles of consolidations</p><a href="http://jco.ascopubs.org/content/29/5/475.full#sec-3">http://jco.ascopubs.org/content/29/5/475.full#sec-3</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable14"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside14">MLL_PTD_coding</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="44">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">MLL_PTD_coding</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p>MLL-PTD have been associated with <b>inferior CR duration and relapse-free survival</b>, although <b>more recent studies show no prognostic impact</b> in patients with CN-AML intensively treated with autologous HSCT or four cycles of consolidations</p><a href="http://jco.ascopubs.org/content/29/5/475.full#sec-3">http://jco.ascopubs.org/content/29/5/475.full#sec-3</a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable15"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside15">ABL1 (exons 4-6)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="47.014925373134%" style="fill: #ffffff"/><defs><clipPath id="45">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="47.014925373134%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="47.014925373134%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="47.014925373134%" font-size="10">exon4</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="50.861079219288%" y="10" height="30" rx="10" ry="10" width="14.638346727899%" style="fill: #ffffff"/><defs><clipPath id="46">
            <rect x="50.861079219288%" y="10" height="30" rx="10" ry="10" width="14.638346727899%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="50.861079219288%" y="10" height="30" rx="10" ry="10" width="14.638346727899%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="50.861079219288%" y="65" width="14.638346727899%" font-size="10">exon5</text><line x1="51.461079219288%" y1="52" x2="51.461079219288%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="67.345579793341%" y="10" height="30" rx="10" ry="10" width="30.654420206659%" style="fill: #ffffff"/><defs><clipPath id="47">
            <rect x="67.345579793341%" y="10" height="30" rx="10" ry="10" width="30.654420206659%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="67.345579793341%" y="10" height="30" rx="10" ry="10" width="30.654420206659%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="67.345579793341%" y="65" width="30.654420206659%" font-size="10">exon6</text><line x1="67.945579793341%" y1="52" x2="67.945579793341%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p></p><a href=""></a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable16"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside16">JAK1_TKD</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="48">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">JAK1_TKD</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p></p><a href=""></a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable17"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside17">JAK2 (Y931)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="66.37856525497%" style="fill: #ffffff"/><defs><clipPath id="49">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="66.37856525497%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="66.37856525497%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="66.37856525497%" font-size="10">Y931</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="70.224719101124%" y="10" height="30" rx="10" ry="10" width="4.6672428694901%" style="fill: #ffffff"/><defs><clipPath id="56">
            <rect x="70.224719101124%" y="10" height="30" rx="10" ry="10" width="4.6672428694901%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="70.224719101124%" y="10" height="30" rx="10" ry="10" width="4.6672428694901%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="70.224719101124%" y="65" width="4.6672428694901%" font-size="10">V617F</text><line x1="70.824719101124%" y1="52" x2="70.824719101124%" y2="43" style="stroke:grey;stroke-width:1" /><rect x="76.738115816767%" y="10" height="30" rx="10" ry="10" width="21.261884183232%" style="fill: #ffffff"/><defs><clipPath id="50">
            <rect x="76.738115816767%" y="10" height="30" rx="10" ry="10" width="21.261884183232%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="76.738115816767%" y="10" height="30" rx="10" ry="10" width="21.261884183232%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="76.738115816767%" y="75" width="21.261884183232%" font-size="10">exon12</text><line x1="77.338115816767%" y1="62" x2="77.338115816767%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p></p><a href=""></a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable18"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside18">BTK (exon15)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="51">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">exon15</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p></p><a href=""></a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable19"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside19">MYD88_mut</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="52">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">MYD88_mut</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p></p><a href=""></a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable20"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside20">PLCG2 (Ser707Tyr)</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="53">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">PLCG2 (Ser707Tyr)</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p></p><a href=""></a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable21"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside21">PML_muts</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="54">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">PML_muts</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p></p><a href=""></a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><table class="reporttable pdfwrapper" id="reporttable22"><tr><td rowspan="6" class="firstcolumn firstcolumnpdf"><div class="hotspotsboxinside badcovered" id="hotspotsboxinside22">SRSF2_c95ex1</div><p class="circlelabel badcovered">No mutations but low coverage in this region</p></td><td><p class="reporttableheadline">Considered Region</p></td></tr><tr>
            <td class="reporttablecontent"><svg width="930px" height="80px" xmlns="http://www.w3.org/2000/svg">
                <line x1="0%" y1="25" x2="100%" y2="25" style="stroke:black;stroke-width:2" /><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/><defs><clipPath id="55">
            <rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="fill: #ffffff"/>
            </clipPath></defs><rect x="2%" y="10" height="30" rx="10" ry="10" width="96%" style="stroke:#000000; fill: none; stroke-width: 1;"/><text x="2%" y="65" width="96%" font-size="10">SRSF2_c95ex1</text><line x1="2.6%" y1="52" x2="2.6%" y2="43" style="stroke:grey;stroke-width:1" /></svg>
            </td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Region evidence</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent"><p></p><a href=""></a></td>
        </tr>
        <tr>
            <td><p class="reporttableheadline">Mutations found</p></td>
        </tr>
        <tr>
            <td class="reporttablecontent">No Mutations</td>
        </tr>
    </table><p style="color:#aaa">Coverage color-coding:</p><svg width="1250px" height="20px" xmlns="http://www.w3.org/2000/svg"><rect x="0" y="1" height="16" rx="3" ry="3" width="16" style="fill: #22670B"/><text x="20" y="16" height="16" width="10%" font-size="10">Full region meets coverage threshold (default: 20 reads)</text><rect x="285" y="1" height="16" rx="3" ry="3" width="16" style="fill: #7EB838"/><text x="305" y="16" height="16" width="10%" font-size="10">>=99% of region meets coverage threshold</text><rect x="512" y="1" height="16" rx="3" ry="3" width="16" style="fill: #B8CF3E"/><text x="532" y="16" height="16" width="10%" font-size="10">>=98% coverage</text><rect x="624" y="1" height="16" rx="3" ry="3" width="16" style="fill: #E3BF3D"/><text x="644" y="16" height="16" width="10%" font-size="10">>=97% coverage</text><rect x="736" y="1" height="16" rx="3" ry="3" width="16" style="fill: #DB7432"/><text x="756" y="16" height="16" width="10%" font-size="10">>=96% coverage</text><rect x="848" y="1" height="16" rx="3" ry="3" width="16" style="fill: #B04925"/><text x="868" y="16" height="16" width="10%" font-size="10">>=95% coverage</text><rect x="960" y="1" height="16" rx="3" ry="3" width="16" style="fill: #902B1D"/><text x="980" y="16" height="16" width="10%" font-size="10">>=94% coverage</text><rect x="1072" y="1" height="16" rx="3" ry="3" width="16" style="fill: #90191C"/><text x="1092" y="16" height="16" width="10%" font-size="10"><94% coverage</text></svg><div class="diagnosispanel"><head4>Possible interpretation</head4><table style="font-size:14px;"><tr><td style="width:400;"><b aria-hidden="true" data-toggle="tooltip" data-placement="top" title="FLT3_ITD=0,NPM1 (type A)=0" style="color:orange;">Intermediate 1 (no NPM1 mutation, no FLT3_ITD)</b></td><td>Source: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5291965/table/T5/?report=objectonly" target="_blank">Modified ELN-Classification</a></td></tr><tr><td style="width:400;"><b aria-hidden="true" data-toggle="tooltip" data-placement="top" title="RUNX1 (exons 3-8)=1" style="color:red;">Adverse (RUNX1 mutation)</b></td><td>Source: <a href="http://iacld.ir/DL/elm/94/acutemyeloidleukemiareviewarticle.pdf" target="_blank">Acute Myeloid Leukemia. Döhner H, Weisdorf DJ, Bloomfield CD. N Engl J Med. 2015 Sep 17;373(12):1136-52.</a></td></tr><tr><td style="width:400;"><b aria-hidden="true" data-toggle="tooltip" data-placement="top" title="KIT (exons 8, 10, 11, 17)=1" style="color:red;">Adverse (c-KIT)</b></td><td>Source: <a href="http://iacld.ir/DL/elm/94/acutemyeloidleukemiareviewarticle.pdf" target="_blank">Acute Myeloid Leukemia. Döhner H, Weisdorf DJ, Bloomfield CD. N Engl J Med. 2015 Sep 17;373(12):1136-52.</a></td></tr></table></div><p style="color:#aaa">This sample was sequenced with lab design 0 (Sequencer: Ilumina, Panel: DEMO).<br>
    The analysis and report generation was generated on Tue, 19 Mar 19 19:04:43  0100 with AMLVaran configuration version 0.<br>
    Detailed information about the processing steps can be obtained from <a href="http://amlvaran.uni-muenster.de/doc/Version0.pdf">http://amlvaran.uni-muenster.de/doc/Version0.pdf.</p>