SELECT va.SampleID, va.SampleName, va.chr, va.pos, va.ref, va.alt, va.Gene, va.Transcripts, ucsc.transcriptEnsemble AS CanonicalTranscript, va.varTypes, va.regionTypes, va.Exons, va.Codons, va.Proteins, va.Impacts, va.Provean_Scores, va.Provean_max, va.SIFT_Scores, va.SIFT_max,
  (va.NRalt/va.Cvg) AS VAF, va.Cvg, va.NRref, va.NRalt, va.BQref, va.BQalt, va.NRref_fwd, va.NRalt_fwd, va.NRref_rev, va.NRalt_rev, va.Cvg_fwd, va.Cvg_rev, va.StrandBias, va.Callers, va.NrCallers,
  a.dbSNP, a.Version AS dbSNP_Version, a.Precious AS dbSNP_PM, 
  Ensembl_transcriptid AS Transcripts_dbNSFP,  SIFT_score AS SIFT_score2, SIFT_pred, SIFT_converted_rankscore AS SIFT_rankscore, b.PROVEAN_score AS PROVEAN_score2, PROVEAN_pred, PROVEAN_converted_rankscore AS PROVEAN_rankscore, Uniprot_acc_Polyphen2, Uniprot_id_Polyphen2, Uniprot_aapos_Polyphen2, Polyphen2_HDIV_score, Polyphen2_HDIV_rankscore, Polyphen2_HDIV_pred, Polyphen2_HVAR_score, Polyphen2_HVAR_rankscore, Polyphen2_HVAR_pred, LRT_score, LRT_converted_rankscore, LRT_pred, LRT_Omega, MutationTaster_score, MutationTaster_converted_rankscore, MutationTaster_pred, MutationTaster_model, MutationTaster_AAE, MutationAssessor_UniprotID, MutationAssessor_variant, MutationAssessor_score, MutationAssessor_score_rankscore, MutationAssessor_pred, FATHMM_score, FATHMM_converted_rankscore, FATHMM_pred, Transcript_id_VEST3, Transcript_var_VEST3, VEST3_score, VEST3_rankscore, MetaSVM_score, MetaSVM_rankscore, MetaSVM_pred, MetaLR_score, MetaLR_rankscore, MetaLR_pred, Reliability_index, `M-CAP_score`, `M-CAP_rankscore`, `M-CAP_pred`, REVEL_score, REVEL_rankscore, MutPred_score, MutPred_rankscore, MutPred_protID, MutPred_AAchange, MutPred_Top5features, CADD_raw, CADD_raw_rankscore, CADD_phred, DANN_score, DANN_rankscore, `fathmm-MKL_coding_score`, `fathmm-MKL_coding_rankscore`, `fathmm-MKL_coding_pred`, `fathmm-MKL_coding_group`, Eigen_coding_or_noncoding, `Eigen-raw`, `Eigen-phred`, `Eigen-PC-raw`, `Eigen-PC-phred`, `Eigen-PC-raw_rankscore`, GenoCanyon_score, GenoCanyon_score_rankscore, integrated_fitCons_score, integrated_fitCons_score_rankscore, integrated_confidence_value, GM12878_fitCons_score, GM12878_fitCons_score_rankscore, GM12878_confidence_value, `H1-hESC_fitCons_score`, `H1-hESC_fitCons_score_rankscore`, `H1-hESC_confidence_value`, HUVEC_fitCons_score, HUVEC_fitCons_score_rankscore, HUVEC_confidence_value, `GERP++_NR`, `GERP++_RS`, `GERP++_RS_rankscore`, phyloP100way_vertebrate, phyloP100way_vertebrate_rankscore, phyloP20way_mammalian, phyloP20way_mammalian_rankscore, phastCons100way_vertebrate, phastCons100way_vertebrate_rankscore, phastCons20way_mammalian, phastCons20way_mammalian_rankscore, SiPhy_29way_pi, SiPhy_29way_logOdds, SiPhy_29way_logOdds_rankscore, TWINSUK_AF, ALSPAC_AF, gnomAD_exomes_AF, gnomAD_exomes_NFE_AF, gnomAD_genomes_AF, gnomAD_genomes_NFE_AF, Interpro_domain, GTEx_V6p_gene, GTEx_V6p_tissue,
  G1000_AF, G1000_eur, d.ESP6500, ExAC_all, ExAC_eur, CLNSIG AS ClinVar_Significance, CLNDN AS ClinVar_Disease, CLNREVSTAT AS Clinvar_Status, CosmicID, CosmicSites, CosmicSNP, NrHaemato AS Cosmic_NrHaemato, NrSamples AS Cosmic_NrSamples, NrSites AS Cosmic_NrSites,
(SELECT COUNT(*) FROM db_civicVariants WHERE chr=va.chr AND va.pos BETWEEN start and stop AND (alt IS NULL OR alt LIKE va.alt)) AS NrCivic,
(SELECT COUNT(DISTINCT Variants.SampleID) FROM Variants LEFT JOIN samples ON Variants.SampleID=samples.SampleID WHERE chr=va.chr AND pos=va.pos AND ref=va.ref AND alt=va.alt AND design=:design AND UserID=:userid AND Variants.SampleID>=3000) AS NrSamples,
(SELECT COUNT(DISTINCT Variants.SampleID) FROM Variants LEFT JOIN samples ON Variants.SampleID=samples.SampleID WHERE chr=va.chr AND pos=va.pos AND design=:design AND UserID=:userid AND Variants.SampleID>=3000) AS NrSamplesSimilar,
(SELECT COUNT(DISTINCT Variants.SampleID) FROM Variants LEFT JOIN samples ON Variants.SampleID=samples.SampleID WHERE chr=va.chr AND pos=va.pos AND ref=va.ref AND alt=va.alt AND (Variants.NRalt/Variants.Cvg)>0.85 AND design=:design AND UserID=:userid AND Variants.SampleID>=3000) AS NrSamplesHighVAF,
min(MutationID) AS inHotspot, va.Category, va.Comment
FROM
  (SELECT v.SampleID, v.SampleName, v.chr, v.pos, v.ref, v.alt, a.Gene, group_concat(a.Transcript) AS Transcripts, group_concat(a.varType) AS varTypes, group_concat(a.regionType) AS regionTypes, group_concat(a.Exon) AS Exons, group_concat(a.Codon) AS Codons, group_concat(a.Protein) AS Proteins, group_concat(a.Impact) AS Impacts, group_concat(IFNULL(a.Provean_Score, "-")) AS Provean_Scores, min(a.Provean_Score) AS Provean_max, group_concat(IFNULL(a.SIFT_Score, "-")) AS SIFT_Scores, min(a.SIFT_Score) AS SIFT_max, v.Cvg, v.NRref, v.NRalt, v.BQref, v.BQalt, v.NRref_fwd, v.NRalt_fwd, v.NRref_rev, v.NRalt_rev, v.Cvg_fwd, v.Cvg_rev, v.StrandBias, v.Callers, v.NrCallers, v.Category, v.Comment FROM
    (SELECT * FROM Variants WHERE SampleID=:sid) AS v
    LEFT JOIN VariantAnno AS a ON v.chr=a.chr AND v.pos=a.pos AND v.ref=a.ref AND v.alt=a.alt
  GROUP BY v.SampleID, v.chr, v.pos, v.ref, v.alt) AS va
  LEFT JOIN db_CanonicalTranscriptsUCSC AS ucsc ON va.Gene=ucsc.GeneSymbol
  LEFT JOIN db_dbSNP    AS a ON va.chr=a.chr AND va.pos=a.pos AND va.ref=a.ref AND va.alt=a.alt
  LEFT JOIN db_dbNSFP   AS b ON va.chr=b.chr AND va.pos=b.pos AND va.ref=b.ref AND va.alt=b.alt
  LEFT JOIN db_G1000    AS c ON va.chr=c.chr AND va.pos=c.pos AND va.ref=c.ref AND va.alt=c.alt
  LEFT JOIN db_ESP6500  AS d ON va.chr=d.chr AND va.pos=d.pos AND va.ref=d.ref AND va.alt=d.alt
  LEFT JOIN db_ExAC     AS e ON va.chr=e.chr AND va.pos=e.pos AND va.ref=e.ref AND va.alt=e.alt
  LEFT JOIN db_clinvar  AS f ON va.chr=f.chr AND va.pos=f.pos AND va.ref=f.ref AND va.alt=f.alt
  LEFT JOIN db_Cosmic   AS g ON va.chr=g.chr AND va.pos=g.pos AND va.ref=g.ref AND va.alt=g.alt
  LEFT JOIN tgt_KnownMutations AS t ON t.version=:version AND va.chr=t.chr AND va.pos BETWEEN t.start AND t.end AND (va.varTypes like concat('%', t.mut_type, '%') OR t.mut_type IS NULL) AND (va.NRalt/va.Cvg)>=t.minVAF
  GROUP BY va.SampleID, va.chr, va.pos, va.ref, va.alt
